src/assignments/launch-checklist.rst

```diff
@@ -0,0 +1,113 @@
+
+.. figure:: figures/form-starting-point.png
+   :alt: Image showing the form and the box stating that more information is needed before launch.
+
+Adding Validation
+-----------------
+
+Adding Alerts
+^^^^^^^^^^^^^
+
+First, let's add the validation to notify the user if they have forgotten to enter a value for any one of the fields.
+
+This process is going to look similar to the :ref:`validation section <javascript-validation>` in the chapter on forms. 
+Make sure to use ``preventDefault()`` and an alert to notify the user that all fields are required.
+
+You also want to make sure that the user has entered valid info for each of the fields.
+Valid information for the fields means that the user has entered a value that is easily converted to the correct data type for our fellow engineers.
+The pilot and co-pilot names should be strings and the fuel level and cargo weight should be numbers.
+
+.. note:: 
+
+   If you want to check if something is ``NaN``, you cannot use ``==`` or ``===``.
+   JavaScript does have a built-in method called ``isNaN(value)`` where it returns ``true`` if ``value`` is ``NaN`` and ``false`` if ``value`` is not ``NaN``.
+
+Updating Shuttle Requirements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The list of shuttle requirements, the ``div`` with the id ``faultyItems``, should be updated if something is not ready for launch. 
+Using template literals, update the ``li`` elements ``pilotStatus`` and ``copilotStatus`` to include the pilot's name and the co-pilot's name.
+
+If the user submits a fuel level that is too low (less than 10,000 liters), change the list to ``visible`` with an updated fuel status stating that there is not enough fuel for the journey.
+The text of the ``h2`` element, ``launchStatus``, should also change to red and "Shuttle not ready for launch".
+
+If the user submits a cargo mass that is too heavy (more than 10,000 kilograms), change the list to ``visible`` with an updated cargo status stating that there is too much mass for the shuttle to take off.
+The text of ``launchStatus`` should also change to red and "Shuttle not ready for launch".
+
+If the shuttle is ready to launch, change the text of ``launchStatus`` to green and display "Shuttle is ready for launch".
+
+Fetching Planetary Data
+-----------------------
+
+Finally, we need some JSON to fill in the crew on the mission destination.
+Our planetary data can be found in `JSON format <https://handlers.education.launchcode.org/static/planets.json/>`_.
+Review the list and decide which planet you want to send our intrepid crew to and make note of the index number.
+
+.. note:: 
+
+   When fetching more than one JSON object, we get an array of all of the JSON objects.
+   In this case, that means an array of our possible mission destinations.
+   So when picking the mission destination, you simply have to pick which item in the array and count starting at 0.
```

Comment:
Grammar suggestion: "When picking the mission destination, just pick the item in the array you want and start counting at 0."

Alternatively, we could ask the students to *randomly select* an item from the array, and provide an internal link to the relevant section in the book.  That way, the students can be surprised as the destination changes!

Reply:
Done! Thanks! I am worried that random selection is too much. Adding it as a Bonus Mission!

Reply:
Works for me.